app/main.ipynb

```diff
@@ -664,6 +1882,7 @@
     "            result.append(modified_num)\n",
     "    return result\n",
     "\n",
+    "\n",
     "my_list = [6, 9, 12, 14, 17, 20]\n",
     "result_python_style = unknown_signature(my_list)\n",
     "result_python_style"
```

Comment:
The NumPy equivalent code for the given Python logic is missing. Implement the logic using NumPy to complete this optional task.